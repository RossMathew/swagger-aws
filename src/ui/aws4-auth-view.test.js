// @flow
import AWS4AuthModel from './aws4-auth-model';
import AWS4AuthView from './aws4-auth-view';

describe('AWS4AuthView', () => {
  it('should not regress', () => {
    const model = new AWS4AuthModel({
      'x-in': 'header',
      'x-service': 'tests',
    });
    const view = new AWS4AuthView({ model });
    expect(view.render().$el.html()).toMatchSnapshot();
  });

  it('highlights invalid fields', () => {
    const model = new AWS4AuthModel({
      'x-in': 'header',
      'x-service': 'tests',
    });
    const view = new AWS4AuthView({ model }).render();

    expect(view.$('.input_aws4_keyId_entry').hasClass('error')).toBe(false);
    expect(view.$('.input_aws4_key_entry').hasClass('error')).toBe(false);

    expect(view.isValid()).toBe(false);
    view.highlightInvalid();
    expect(view.$('.input_aws4_keyId_entry').hasClass('error')).toBe(true);
    expect(view.$('.input_aws4_key_entry').hasClass('error')).toBe(true);

    model.set('keyId', 'theKeyId');
    model.set('key', 'theKey');
    expect(view.isValid()).toBe(true);
  });

  /* This currently doesn't work... Neither does it in the standard SwaggerUI
   * components */
  xit('removes highlighting of invalid fields after input entered', () => {
    const model = new AWS4AuthModel({
      'x-service': 'tests',
    });
    const view = new AWS4AuthView({ model }).render();

    view.highlightInvalid();
    expect(view.$('.input_aws4_keyId_entry').hasClass('error')).toBe(true);
    expect(view.$('.input_aws4_key_entry').hasClass('error')).toBe(true);

    model.set({
      keyId: 'theKeyId',
      key: 'theKey',
    });

    view.highlightInvalid();
    expect(view.$('.input_aws4_keyId_entry').hasClass('error')).toBe(false);
    expect(view.$('.input_aws4_key_entry').hasClass('error')).toBe(false);
  });
});
