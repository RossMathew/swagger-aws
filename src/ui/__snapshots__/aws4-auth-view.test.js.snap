// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AWS4AuthView should not regress 1`] = `
"<div class=\\"key_input_container\\">
  <h3 class=\\"auth__title\\">AWS4 authorization</h3>
  <div class=\\"auth__description\\"></div>
  <div>
    <!--
    <div class=\\"key_auth__field\\">
      <span class=\\"key_auth__label\\">in:</span>
      <span class=\\"key_auth__value\\">header</span>
    </div>
    -->
    <div class=\\"key_auth__field\\">
      <span class=\\"key_auth__label\\">Service:</span>
      <span class=\\"key_auth__value\\">tests</span>
    </div>
    <div class=\\"key_auth__field\\">
      <span class=\\"key_auth__label\\">Region:</span>
      <span class=\\"key_auth__value\\">be-west-1</span>
    </div>
    <div>
      <span class=\\"key_auth__label\\">Key ID:</span>
        <input required=\\"\\" placeholder=\\"Key ID\\" class=\\"auth_input input_aws4_keyId_entry\\" name=\\"keyId\\" type=\\"text\\">
    </div>
      <div>
        <span class=\\"key_auth__label\\">Key:</span>
        <input required=\\"\\" placeholder=\\"Key\\" class=\\"auth_input input_aws4_key_entry\\" name=\\"key\\" type=\\"password\\">
      </div>
  </div>
</div>
"
`;
